repo/blob/azure/azure_storage.go

```diff
@@ -168,7 +234,12 @@ func (az *azStorage) getObjectNameString(b blob.ID) string {
 
 // ListBlobs list azure blobs with given prefix.
 func (az *azStorage) ListBlobs(ctx context.Context, prefix blob.ID, callback func(blob.Metadata) error) error {
-	prefixStr := az.Prefix + string(prefix)
+	prefixStr := az.getObjectNameString(prefix)
+
+	blobsToSkip, err := az.getBlobsToSkip(ctx, prefix)
```

Comment:
A side note, it would be possible to skip this step if the user hasn't set retention period. So the behaviour will be unchanged for users not taking the immutability path (1 list call instead of 2)

Reply:
extra step is skipped